core/source/Number.mint

```diff
@@ -42,9 +42,13 @@ module Number {
   fun fromString (input : String) : Maybe(Number) {
     `
     (() => {
-      let value = parseFloat(#{input})
+      let value = Number(#{input})
 
-      if (isNaN(value)) {
+      if (
+        Number.isNaN(value)
+        ||
+        Number.isNaN(value - parseFloat(#{input}))  // excludes "Infinity", "-Infinity", ""
```

Comment:
Why not compare with `Infinity` and `-Infinity`?

Reply:
I was actually asking this myself too. I am open to including it, but I am unsure about parsing `""` as `0`. It may result in problems when doing `Validate.isNumber()`.

Reply:
Why not leave `parseFloat`?

Reply:
@nilslindemann provided a reason in another comment https://github.com/mint-lang/mint/pull/612#discussion_r1210172933, but basically `8O` would be `8` instead of `80` thus can cause confusion.